### ISSUE TITLE
Passing a frozen channel name or name gives an error on the REST client

### ISSUE THREAD

**Issue by aschuster3** [ably/ably-ruby#132]
While using the REST client, I started getting a `RuntimeError: can't modify frozen String`

Using the minimum reproduction:
```ruby
# example.rb
require 'ably'

# API Key redacted, but I have tested using a valid key
ably_client = Ably::Rest.new('')

a_frozen_channel_name = 'channel_name'.freeze
ably_client.channel(a_frozen_channel_name)
```
I get the following error trace:
```
$HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/modules/conversions.rb:99:in `encode!': can't modify frozen String (RuntimeError)
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/modules/conversions.rb:99:in `ensure_utf_8'
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/rest/channel.rb:25:in `initialize'
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/modules/channels_collection.rb:26:in `new'
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/modules/channels_collection.rb:26:in `get'
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/rest/channels.rb:18:in `get'
	from $HOME/.gem/ruby/2.3.3/gems/ably-rest-1.0.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:204:in `channel'
	from example.rb:6:in `<main>'
```

To save the hassle, [here's the link to the offending line of code](https://github.com/ably/ably-ruby/blob/master/lib/ably/modules/conversions.rb#L99)

---

A workaround I've found is to call `.dup` on the frozen string (which just copies the string and gives you a mutable version of it).
```ruby
# workaround.rb
require 'ably'

# API Key redacted, but I have tested using a valid key
ably_client = Ably::Rest.new('')

a_frozen_channel_name = 'channel_name'.freeze

# call .dup on our channel name
ably_client.channel(a_frozen_channel_name.dup)
```

Ably version:
```ruby
irb(main):001:0> Ably::VERSION
=> "1.0.4"
```

Ruby version:
```sh
❯ ruby -v
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin16]
```

**Comment by funkyboy** [comment]
Thanks for spotting this @aschuster3! 
Would you like to make a PR so that we can test your proposed change on CI and then review it?

**Comment by thadeu** [comment]
guys, I'm getting same error when publish in the channel...

**Comment by mattheworiordan** [comment]
@thadeu can you please confirm Ruby version and version of the library, and the error and stacktrace please.

**Comment by thadeu** [comment]
@mattheworiordan Yes, sure.

**ruby version**

```
ruby 2.3.6p384 (2017-12-14 revision 61254) [x86_64-darwin17]
```

**lib version**

```
* ably (1.0.5)
```

if sounds good, I'll getting open a issue for this. If you think about?

**Comment by mattheworiordan** [comment]
@thadeu please see https://github.com/ably/ably-ruby/commit/afdcc6462e54ed40572e3e00ab1253b50706d9c1.  I cannot reproduce this with 1.1.  Would you mind trying to upgrade to 1.1 to see if the issue still exists?

**Comment by TheSmartnik** [comment]
@mattheworiordan @thadeu hi! I'm closing this since the fix was released in [1.0.6](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md#v106-2018-05-01) and is expected to fail in 1.0.5
